runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -1347,22 +1796,15 @@ TR::Register *J9::Power::TreeEvaluator::irdbariEvaluator(TR::Node *node, TR::Cod
    cg->stopUsingRegister(condReg);
 
    cg->machine()->setLinkRegisterKilled(true);
-
-   return objReg;
-#else
-   return NULL;
+   return objReg;   
 #endif
    }
 
-TR::Register *J9::Power::TreeEvaluator::ardbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-   {
-   return TR::TreeEvaluator::ardbariEvaluator(node, cg);
-   }
-
-TR::Register *J9::Power::TreeEvaluator::ardbariEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+TR::Register *aConcurrentScavengeSequence(TR::Node *node, TR::CodeGenerator *cg)
    {
-   TR_ASSERT(TR::Compiler->om.readBarrierType() != gc_modron_readbar_none, "aReadBarrierEvaluator");
-#ifdef OMR_GC_CONCURRENT_SCAVENGER
+#ifndef OMR_GC_CONCURRENT_SCAVENGER
+   TR_ASSERT_FATAL(0, "Concurrent Scavenger not supported.");
```

Comment:
I don't find this assert necessary. What is it supposed to guard against?

Reply:
This path should not be executed during runtime on 32 bit machines, since CS is not supported. This is simply just a safety to make sure we don't end up down this path.

Reply:
> I think we need to return NULL here in order for things to compile.

That makes sense.